modules/ting_new_materials/ting_new_materials.module

```diff
@@ -150,25 +153,22 @@ function ting_new_materials_make_limit_query($limit = 0) {
  * @param int $num_per_page
  *   The page number.
  *
- * @return TingClientSearchResult
+ * @return TingSearchResultInterface
  *   The search result.
  */
 function ting_new_materials_do_search($query, $page = 0, $num_per_page = 10) {
-  // Load the ting client.
-  module_load_include('client.inc', 'opensearch');
-
-  // Build search options and ensure that they are sorted correctly.
-  $options = array();
-  //$options['sort'] = 'acquisitionDate_descending';
-  $options['sort'] = 'date_descending';
-  $options['enrich'] = TRUE;
-  $options['collectionType'] = 'manifestation';
-
   // Search the data well.
   $search_result = array();
   try {
-    // TODO: Consider porting this to the Search Abstraction Layer.
-    $search_result = opensearch_do_search($query, $page + 1, $num_per_page, $options);
+    $sal_query = ting_start_query()
+      ->setFullTextQuery($query)
+      ->setPage($page + 1)
+      ->setCount($num_per_page)
+      ->addSort('date', TingSearchSort::DIRECTION_DESCENDING)
+      ->addFieldFilter('enrich', TRUE)
+      ->addFieldFilter('collectionType', 'manifestation');
```

Comment:
As mentioned in another PR: Are `enrich` and `collectionType` valid field filters?